sql/src/main/java/org/apache/druid/sql/calcite/rel/DruidOuterQueryRel.java

```diff
@@ -129,8 +131,10 @@ public DruidQuery toDruidQuery(final boolean finalizeAggregations)
     }
 
     final RowSignature sourceRowSignature = subQuery.getOutputRowSignature();
+    final GroupByQuery groupByQuery = subQuery.toGroupByQuery();
+    final DataSource dataSource = groupByQuery == null ? subQuery.getDataSource() : new QueryDataSource(groupByQuery);
```

Comment:
I don't think this is correct. It is ignoring everything about `subQuery` except the underlying dataSource. It means that if the underlying query is doing something "interesting", like a limit, or expression, etc, then it will be ignored.

I'd suggest doing this instead, to at least return a "nice" null instead of throwing NPE. Then that will get understood by the planner as an unplannable query.

```java
if (groupByQuery == null) {
   return null;
}
```

Reply:
Yep, this modification seems to be better. Thanks.